crates/holochain_state/src/validation_receipts.rs

```diff
@@ -69,26 +71,27 @@ impl ValidationReceipt {
                 async move { validator.sign(&keystore, this).await }
             })
             .collect::<Vec<_>>();
-        let signatures = futures::stream::iter(futures)
-            .buffer_unordered(10)
-            .collect::<Vec<_>>()
-            .await;
-        let signatures = if signatures.iter().any(Result::is_ok) {
-            signatures.into_iter().filter_map(Result::ok).collect()
-        } else {
-            return Err(signatures
-                .into_iter()
-                .filter_map(Result::err)
-                .next()
-                .expect("Must contain at least one error if there is no success"));
```

Comment:
do i understand this correctly that the semantics of this method has changed and the responsibility of treating this condition as an error is now with the caller? if so, i think it would be worth adding this as a note to the docstring of the method.

Reply:
I've added the expect back because the tests verify that this expect can't be hit now so the semantics are now the same.